project-base/storefront/components/Pages/Order/ContactInformation/contactInformationFormMeta.ts

```diff
@@ -56,55 +58,99 @@ export const useContactInformationForm = (): [UseFormReturn<ContactInformation>,
                 otherwise: Yup.string(),
             }),
             differentDeliveryAddress: Yup.boolean(),
-            deliveryFirstName: Yup.string().when('differentDeliveryAddress', {
-                is: true,
+            deliveryFirstName: Yup.string().when(['differentDeliveryAddress', 'deliveryAddressUuid'], {
+                is: (differentDeliveryAddress: boolean, deliveryAddressUuid: string) =>
```

Comment:
Wouldn't be better naming such as `newDeliveryAddress`? Just by looking I have no idea what does `different` mean.

Reply:
Yeah, I see how that is confused. However, this is how it is named on the API. It is a boolean parameter saying if the delivery address is different from billing address, which means, that the name is actually so confusing, that it was even hard for you to see what the parameter is for, for which I don't blame you 😆 I created a [US](https://shopsys.atlassian.net/browse/SSP-2212) where we rename this from `differentDeliveryAddress` to `isDeliveryAddressDifferentFromBilling`, ok?

Reply:
Ah, I see. Sure! Thanks.